pkg/pipelines/component/types.go

```diff
@@ -0,0 +1,16 @@
+package pipelines
+
+type GeneratorOptions struct {
+	Output               string //
```

Comment:
@JainamShah-08 , in a separate Jira/PR, clean up all the comments. We will have to fill these in.  

I need your changes in now since they depend on your changes.